spec/work_ease_spec.rb

```diff
@@ -0,0 +1,289 @@
+#! /usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative '../workease2'
+require 'timecop'
+
+bodypart_activity = {
+  feet: { last_activity: nil,
+          activity_level: 0,
+          min_rest: 5,
+          max_exertion: 19,
+          high_activity_start: nil },
+  hands: { last_activity: nil,
+           min_rest: 5,
+           activity_level: 0,
+           max_exertion: 10,
+           high_activity_start: nil },
+  voice: { last_activity: nil,
+           min_rest: 10,
+           activity_level: 0,
+           max_exertion: 20,
+           high_activity_start: nil }
+}
+
+keyboard_id, mouse_id = WorkEase.find_device_ids(keyboard_name: 'VirtualBox USB Keyboard', mouse_name: 'VirtualBox mouse integration')
+
+RSpec.describe WorkEase do
+  before(:each) do
+    @w = WorkEase.new
+    @w.testing = true
+    @time = Time.at(1591192757)
+  end
+
+  after(:each) do
+    Timecop.return
+  end
+
+  def use(bodypart, bodypart_activity, time)
+    bodypart_activity[bodypart][:last_activity] = time.to_i
+    @w.bodypart = bodypart_activity
```

Comment:
The name of this method is `use`, and further along in this file the method is used to simulate activity. However, on line 40 you overwrite the complete state with the initial state. And in fact, on the previous line where you overwrite the `:last_activity`, you may actually be overwriting the initial state used in other tests.

I think it would be better to either rewrite this helper to not overwrite the initial state or, even better: don't use this helper at all. Workease already has a `check` method that is called upon activity, I would use that same method from the test as well. That means your test fixture will likely contain a few more warnings, but that seems worth it to me.

Reply:
Using the check method from workease instead of this helper everywhere works right out of the gate, no need to change any fixtures for this.